src/relay/analysis/annotated_region_set.cc

```diff
@@ -131,7 +131,13 @@ class AnnotatedRegionSet::Creator : public ExprVisitor {
         CHECK_EQ(region->GetTarget(), target);
       }
       region->nodes_.insert(GetRef<Call>(call));
-      region->outs_.push_back(GetRef<Call>(call));
+      if (!std::any_of(region->outs_.begin(), region->outs_.end(),
```

Comment:
You probably need to add this logic to `MergeRegion`, too. Since this logic should be used every time we want to manipulate region outputs, I'd prefer to have an API like `Region::AddOutput` and make `outs_` private.